ui/src/song/SongList.js

```diff
@@ -78,15 +78,10 @@ const SongList = (props) => {
         perPage={isXsmall ? 50 : 15}
       >
         {isXsmall ? (
-          <SimpleList
-            primaryText={(r) => r.title}
-            secondaryText={(r) => r.artist}
-            tertiaryText={(r) => (
-              <>
-                <DurationField record={r} source={'duration'} />
-                &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
-              </>
-            )}
+          <SongSimpleList
```

Comment:
If it is a _Song_ SimpleList, why not just passing the whole song record to it (as a `record` property), and encapsulate all complexities of the component?

Reply:
actually, the title is not getting passed as a plain text prop here, it is actually giving the title back whenever the title function is called in the SongSimpleList component. yes, we can remove it and implement it in the component only, but actually, it is useful to have it we can control the song properties like if we wanted to append the extra artists to the song title like (**ft.ARTIST_NAME**) we could implement it easily.

also having this type of props is super useful for scalability, we don't need to dig up the whole songsimplelist component to just access some properties of the record.

Reply:
if you suggest implementing it inside the component itself I can do it though :)

Reply:
> but actually, it is useful to have it we can control the song properties like if we wanted to append the extra artists to the song title like (**ft.ARTIST_NAME**) we could implement it easily.

Actually this is a good argument to NOT have it as individual properties: If we reuse this component in another place (after all, the point of using components is for reusability and encapsulation), and we decide to change the song title, we would have to change the `title` property in all places the component is being used, risking having them differently in different contexts. If we just pass the record and let the component handle the logic, we can just use it knowing it will do the right thing.

Please change it to receive only `record`, if we need a flexible component, then we can use the original SimpleList, that is customizable.

Reply:
Ok :)